src/ExpressionParser.php

```diff
@@ -376,8 +377,18 @@ public function parseHashExpression()
                 throw new SyntaxError(sprintf('A hash key must be a quoted string, a number, a name, or an expression enclosed in parentheses (unexpected token "%s" of value "%s".', Token::typeToEnglish($current->getType()), $current->getValue()), $current->getLine(), $stream->getSourceContext());
             }
 
-            $stream->expect(/* Token::PUNCTUATION_TYPE */ 9, ':', 'A hash key must be followed by a colon (:)');
-            $value = $this->parseExpression();
+            // a hash can be shorten when the key and value got the same identifier ({ foo: foo, 'bar': 'bar' } === { foo, 'bar' })
+            if ($stream->nextIf(/* Token::PUNCTUATION_TYPE */ 9, ':')) {
+                $value = $this->parseExpression();
+            } else if ($value === null) {
```

Comment:
CS: `} elseif (null === $value) {` (notice the removed whitespace between `else` and `if`) and the inversion of `null` and `$value`